sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2589,6 +2589,8 @@ object Sequence {
     }
   }
 
+  // To generate time sequences, we use scale 1 in TemporalSequenceImpl
+  // for `TimestampType`, while MICROS_PER_DAY for `DateType`
```

Comment:
if start/end is date, can the step by seconds/minutes/hours?

Reply:
Yes, we can, but the result seems a little strange：
`scala> sql("select explode(sequence(cast('2011-03-01' as date), cast('2011-05-01' as date), interval 1 hour))").count
res19: Long = 1465

scala> sql("select explode(sequence(cast('2011-03-01' as date), cast('2011-05-01' as date), interval 1 minute))").count
res20: Long = 87841

scala> sql("select explode(sequence(cast('2011-03-01' as date), cast('2011-05-01' as date), interval 1 second))").count
res21: Long = 5270401`

Reply:
does pgsql support it?

Reply:
Seems pgsql can only support int as follows：
postgres= create sequence seq_test;
CREATE SEQUENCE
postgres= select nextval('seq_test');
       1
(1 行记录)
postgres= select nextval('seq_test');
       2
(1 行记录)

Reply:
Actually this function is from presto: https://prestodb.io/docs/current/functions/array.html

Can you check the behavior of presto? It looks confusing to use time fields as the step for date start/stop.

Reply:
presto> select sequence(date('2011-03-01'),date('2011-03-02'),interval '1' hour);
Query 20200624_122744_00002_pehix failed: sequence step must be a day interval if start and end values are dates
presto> select sequence(date('2011-03-01'),date('2011-03-02'),interval '1' day);
          _col0           
 [2011-03-01, 2011-03-02] 
(1 row)
Query 20200624_122757_00003_pehix, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]
presto> select sequence(date('2011-03-01'),date('2011-03-02'),interval '1' month);
    _col0     
 [2011-03-01] 
(1 row)

Query 20200624_122806_00004_pehix, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]
presto> select sequence(date('2011-03-01'),date('2011-03-02'),interval '1' year);
    _col0     
 [2011-03-01] 
(1 row)
Query 20200624_122810_00005_pehix, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

Reply:
@cloud-fan Done, It seems can be sequence `day`,`month`,'year' when start and end are date type.

Reply:
I think the presto behavior makes sense. Can we send a PR to follow it first? e.g. throw an exception if the step is time fields while start/end is date. This can also simplify the implementation.

Reply:
Ok, I will do it tomorrow.